### DISCUSSION TITLE
Logs say sending out email but user not really receiving any in their inbox

### DISCUSSION THREAD

**Discussion by jaya-kumar-softobiz** [ory/kratos#2105] [Q&A] [answered]
After upgrading ory version to 0.8, the mails are not being sent at all to the users. It worked fine in 0.5.5
Tried these
connection_uri: smtp://apikey:SG.tyomYyLDThq4r-pq5Xhy6w.OpQhRa4R0iDl3uZKbpsj@smtp.sendgrid.net:587/?skip_ssl_verify=false (This worked fine in 0.5.5)

connection_uri: smtp://apikey:SG.tyomYyLDThq4r-pq5Xhy6w.OpQhRa4R0iDl3uZKbpsj@smtp.sendgrid.net:587/?disable_starttls=true&?skip_ssl_verify=false

Front end and backend are both on https. Using send grid account, as you can see. 
Anything changed in 0.8 ? DO I have to add some flag or something? Please help. Thank you

here is the log exactly when it says sending out email
{
  audience | audit
  email_address | Value is sensitive and has been redacted. To see the value set config key
  identity_id | 2bea38b9-0640-4d73-8264-d7f1d9a2f856
  level | info
  msg | Sending out recovery email with recovery link.
  recovery_link_id | c38bdf1a-b4bd-4606-819e-9325a7e2cf72
  recovery_link_token | Value is sensitive and has been redacted. To see the value set config key
  service_name | Ory Kratos
  service_version | v0.8.0-alpha.3
  time | 2022-01-04T15:37:15Z
  via | email
}

**Comment by aeneasr** [comment, marked as answer]
Please read the upgrade guide for 0.8 and 0.7 (and any other upgrade you perform). It includes vital information for the upgrade process :)

https://github.com/ory/kratos/blob/master/CHANGELOG.md

**Reply by jaya-kumar-softobiz** [reply]
Ok cool, thank you so much. I did check it, but will go through it more thoroughly once again

**Reply by jaya-kumar-softobiz** [reply]
Also, I forgot to tell you that it works fine in my local env using 127.0.0.1
It is not working in our QA or prod( which are secured https) urls
When I asked "Anything changed in 0.8",  Obviously a lot changed in 0.8... but I meant to ask regarding smtp connection uri config.

**Reply by aeneasr** [reply]
It is probably related to the SSL changes we made to the courier URL format

**Reply by jaya-kumar-softobiz** [reply]
Oh got it, any info or links please

**Reply by aeneasr** [reply]
It‘s in the linked changelog and in the docs when you search for smtp :)

**Reply by jaya-kumar-softobiz** [reply]
Awesome, thank you very much. Cool project by the way

**Reply by aeneasr** [reply]
You’re welcome :) If this helped can you please mark this answer as the solution?

**Reply by himanshuZee** [reply]
@jaya-kumar-softobiz We have integrated ory kratos in our org. We are facing the same issue that email is not delivered to the users. We have also configured sendgrid the way you did.
Can you please help us which courier.connection_uri worked for you ??

Note : We are using the v0.10.1 version of kratos.
Thanks
Himanshu

**Reply by jaya-kumar-softobiz** [reply]
Hey, it didn't work because we upgraded from 0.5 to 0.8 at once. Instead, when we did 0.5 to 0.6 -> 0.7 -> 0.8, it started working. But I have no idea what's the reason
This URI worked 
connection_uri: smtp://apikey:YourApiKey@smtp.sendgrid.net:587/?disable_starttls=true&?skip_ssl_verify=false

Also, try without the / after 587 and try removing both query params or one at a time and see

**Reply by vinckr** [reply]
By the way, if you don't want to deal with upgrading and maintaining, check out the Ory Network (https://console.ory.sh/). There we handle pesky things like upgrades, latency and so on and you can concentrate on your work.

**Reply by himanshuZee** [reply]
@jaya-kumar-softobiz  Thanks for the reply. Trying out the configs but no luck.
